sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -300,13 +298,17 @@ func (m *networkFlowManager) ProcessMessage(_ *central.MsgToSensor) error {
 
 func (m *networkFlowManager) Start() error {
 	go m.enrichConnections(m.enricherTicker.C)
-	go m.publicIPs.Run(&m.done, m.clusterEntities)
+	go m.publicIPs.Run(m.stopper.LowLevel().GetStopRequestSignal(), m.clusterEntities)
 	return nil
 }
 
 func (m *networkFlowManager) Stop(_ error) {
-	m.done.Signal()
-	m.finished.Wait()
+	if !m.stopper.Client().Stopped().IsDone() {
+		defer func() {
+			_ = m.stopper.Client().Stopped().Wait()
+		}()
+	}
+	m.stopper.Client().Stop()
```

Comment:
Why do we need this if-clause with a defer? Would the following not work? 
```suggestion
	m.stopper.Client().Stop()
	_ = m.stopper.Client().Stopped().Wait()
```

It feels weird to wait only if something is not stopped. As we are always stopping the component in 311, then we can simply wait directly afterwards.

Reply:
You are right the code you suggest would work. I've only added the `if` and the `defer` for consistency with other components, but there's no real value here.